modules/javafx.controls/src/test/java/test/javafx/scene/control/TooltipTest.java

```diff
@@ -542,16 +555,155 @@ public class TooltipTest {
-        Rectangle rect1 = new Rectangle(0, 0, 100, 100);
+        Rectangle rect = new Rectangle(0, 0, 100, 100);
+
+        stageLoader = new StageLoader(rect);
+
+        Tooltip.install(rect, toolTip);
+
+        MouseEvent mouseEvent = MouseEventGenerator.generateMouseEvent(MouseEvent.MOUSE_MOVED, 1, 1);
+        rect.fireEvent(mouseEvent);
+    }
+
+    @Test
+    public void testTooltipShouldNotBeShownBeforeDefaultDelay() {
+        Rectangle rect = new Rectangle(0, 0, 100, 100);
+
+        stageLoader = new StageLoader(rect);
+
+        Tooltip.install(rect, toolTip);
+
+        MouseEvent mouseEvent = MouseEventGenerator.generateMouseEvent(MouseEvent.MOUSE_MOVED, 1, 1);
+        rect.fireEvent(mouseEvent);
+
+        assertFalse(toolTip.isShowing());
+
+        toolkit.setAnimationTime(999);
+
+        assertFalse(toolTip.isShowing());
+    }
+
+    @Test
+    public void testTooltipShouldBeShownAfterDefaultDelay() {
+        Rectangle rect = new Rectangle(0, 0, 100, 100);
+
+        stageLoader = new StageLoader(rect);
+
+        Tooltip.install(rect, toolTip);
+
+        assertFalse(toolTip.isShowing());
+
+        assertTooltipShownAfter(rect, 1000);
+        assertTooltipHiddenAfter(rect, 200);
+    }
+
+    @Test
+    public void testTooltipShouldBeHiddenAfterHideDelay() {
+        int delay = 50;
+        toolTip.setHideDelay(Duration.millis(delay));
+
+        Rectangle rect = new Rectangle(0, 0, 100, 100);
+
+        stageLoader = new StageLoader(rect);
+
+        Tooltip.install(rect, toolTip);
+
+        assertFalse(toolTip.isShowing());
+
+        assertTooltipShownAfter(rect, 1000);
+        assertTooltipHiddenAfter(rect, delay);
+    }
+
+    @Test
+    public void testTooltipShouldBeShownAfterSetShowDelay() {
+        int delay = 200;
+        toolTip.setShowDelay(Duration.millis(delay));
+
+        Rectangle rect = new Rectangle(0, 0, 100, 100);
+
+        stageLoader = new StageLoader(rect);
+
+        Tooltip.install(rect, toolTip);
+
+        assertFalse(toolTip.isShowing());
+
+        assertTooltipShownAfter(rect, delay);
+        assertTooltipHiddenAfter(rect, 200);
+    }
+
+    @Test
+    public void testTooltipShouldBeShownAfterSetStyleShowDelay() {
+        toolTip.setStyle("-fx-show-delay: 200ms;");
+
+        Rectangle rect = new Rectangle(0, 0, 100, 100);
+
+        stageLoader = new StageLoader(rect);
+
+        Tooltip.install(rect, toolTip);
+
+        assertFalse(toolTip.isShowing());
+
+        assertTooltipShownAfter(rect, 200);
+        assertTooltipHiddenAfter(rect, 200);
+    }
+
+    @Test
+    public void testTooltipShouldBeShownAfterSetCssShowDelay() {
+        Rectangle rect = new Rectangle(0, 0, 100, 100);
+
+        stageLoader = new StageLoader(rect);
+        // Style: .tooltip { -fx-show-delay: 200ms; }
+        stageLoader.getStage().getScene().getStylesheets()
+                .add("data:base64,LnRvb2x0aXAgeyAtZngtc2hvdy1kZWxheTogMjAwbXM7IH0=");
```

Comment:
this is clever.  can we generate the base64-encoded string on the fly?  something along the lines

```
return "data:text/css;base64," + Base64.getEncoder().encodeToString(b);
```

(here and elsewhere?)

Reply:
Thought about this well, didn't test yet

Reply:
see for example 
https://github.com/openjdk/jfx/blob/72701e6cb4095b8f5042f54ae6bb2c0cec446bcf/modules/javafx.graphics/src/test/java/test/javafx/scene/Node_transition_Test.java#L142

Reply:
Ah, looks good. Maybe in the future we can create a Utility function for this.